patches/minecraft/net/minecraft/world/server/ChunkManager.java.patch

```diff
@@ -8,23 +8,58 @@
                 }
  
                 this.func_219229_a(p_219185_5_);
-@@ -601,6 +602,7 @@
-                if (list != null) {
-                   list.forEach(chunk::func_76622_b);
-                }
-+               net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.ChunkEvent.Load(chunk));
+@@ -582,25 +583,28 @@
+             if (this.field_219254_h.add(chunkpos.func_201841_a())) {
+                chunk.func_177417_c(true);
+                this.field_219255_i.func_147448_a(chunk.func_177434_r().values());
+-               List<Entity> list = null;
+-               ClassInheritanceMultiMap<Entity>[] aclassinheritancemultimap = chunk.func_177429_s();
+-               int i = aclassinheritancemultimap.length;
++               p_219200_1_.func_219276_a(ChunkStatus.field_222617_m, this).thenRun(() -> { //Forge: Delay EntityJoinWorld events until chunk is available to prevent deadlocks
++                  List<Entity> list = null;
++                  ClassInheritanceMultiMap<Entity>[] aclassinheritancemultimap = chunk.func_177429_s();
++                  int i = aclassinheritancemultimap.length;
+ 
+-               for(int j = 0; j < i; ++j) {
+-                  for(Entity entity : aclassinheritancemultimap[j]) {
+-                     if (!(entity instanceof PlayerEntity) && !this.field_219255_i.func_217440_f(entity)) {
+-                        if (list == null) {
+-                           list = Lists.newArrayList(entity);
+-                        } else {
+-                           list.add(entity);
++                  for(int j = 0; j < i; ++j) {
++                     for(Entity entity : aclassinheritancemultimap[j]) {
++                        if (!(entity instanceof PlayerEntity) && !this.field_219255_i.func_217440_f(entity)) {
++                           if (list == null) {
++                              list = Lists.newArrayList(entity);
++                           } else {
++                              list.add(entity);
++                           }
+                         }
+                      }
+                   }
+-               }
+ 
+-               if (list != null) {
+-                  list.forEach(chunk::func_76622_b);
+-               }
++                  if (list != null) {
++                     list.forEach(chunk::func_76622_b);
++                  }
```

Comment:
If you don't change the indentation of this inner stuff the patch will be a lot cleaner/smaller.

Reply:
Changed and pushed

Reply:
I think your change of this might have lost something? The description of the PR says you moved the load event, but it seems you are only adding a new one now, instead of also removing the old one.

Reply:
I think your reading the patch wrong. There's only one post of `ChunkEvent.Load`?

Reply:
Ya nevermind I was reading it wrong, because there was an extra line I think just because of all the lines earlier that I think was showing removed.

Reply:
Literally woke up in the middle of the night and realised I missed a line from my fix. I’ll fix this in the morning (If I’m actually right)